Dockerfile

```diff
@@ -9,8 +9,31 @@ RUN export DEBIAN_FRONTEND="noninteractive" && apt-get update && \
     apt-get install -y --no-install-recommends \
       kicad kicad-libraries zip inkscape make git libmagickwand-dev \
       python3 python3-pip python3-wheel python3-setuptools inkscape \
-      libgraphicsmagick1-dev libmagickcore-dev openscad
+      libgraphicsmagick1-dev libmagickcore-dev openscad && \
+      rm -rf /var/lib/apt/lists/*
 
-RUN pip3 install Pcbdraw KiKit
+RUN pip3 install Pcbdraw
 
-CMD ["bash"]
\ No newline at end of file
+# hack: manually install Python dependencies to speed up the build process
+# for repetitive builds
+
+RUN pip3 install numpy shapely click markdown2 pybars3 solidpython
+
+# create a new stage for building and installing KiKit
+FROM base AS build
+
+COPY . /src/kikit
+WORKDIR /src/kikit
+RUN python3 setup.py install
+
+# the final stage only takes the installed packages from dist-packages 
+# and ignores the src directories
+FROM base
+COPY --from=build \
+    /usr/local/lib/python3.8/dist-packages \
+    /usr/local/lib/python3.8/dist-packages
+COPY --from=build \
+    /usr/local/bin \
+    /usr/local/bin
```

Comment:
I have one note here for discussion; currently, only binaries are copied. However, that
- makes the installation of `kicad-libraries` useless, as it installs files under `/usr/share`.
- Similarly, it does not copy KiCAD resource files (e.g., translations or icons). Therefore, the KiCAD cannot be no longer run from the container.

The question is about the usage of the final image. I know some users, that run KiCAD (with an external X server) from the container. Originally, I included the libraries so people can programmatically, e.g., add footprints in CI via scripting. This, however, bloats the image, makes pulling the image in CI slower, and also makes the incremental builds much slower. So I am not sure where to put the boundary what we should include/exclude.

Any opinions?

Reply:
Everything in the `base` stage is still available in the final stage. Note that the final stage is based on `base`. Only the intermediate Python build files from the `build` stage are thrown away. (Or at least that was the intention!) I've only used this Dockerfile with the basic use case of running KiKit CLI via Docker but I think running KiCAD itself should still work. Such a use case never ever occurred in my mind, though.

I don't have a strong opinion on what the Docker image should contain. I guess I don't really know enough about KiCAD packaging to have an enlightened discussion on the topic. :-) Based on 10 seconds of research, it seems to me like the `kicad` package provides both the CLI tools and the GUI, right? The GUI parts must pull quite a bit of packages in, so the image size is going to be quite large in any case; especially with Inkscape and OpenSCAD. (I suppose they're required for some of KiKit's functionality, right?) I'm not sure what size reduction possibilities there are.

Reply:
As in the previous comment, I misread the Dockerfile. It makes sense and it works as expected. So this note is no longer relevant.

Inkscape is included as an optional dependency for PcbDraw used in the present module. Pcbdraw uses it indirectly via libmagick to convert SVG files into raster images. The build-in SVG converters in libmagick unfortunately often yield raster images with artifacts.